### ISSUE TITLE
Expose form status types (VALID, INVALID, PENDING, DISABLED)

### ISSUE THREAD

**Issue by VagrantAI-c** [angular/angular#27389]
# 🚀 feature request

### Relevant Package

 This feature request is for @angular/forms

### Description
 It would be great if form status types (such as `VALID`, `INVALID`, `PENDING `and `DISABLED`) would be exposed to public api

### Describe the solution you'd like
Comparing form statuses to static strings eliminates Typescript as a concept, typechecking could benefit if these consts (which are already defined in `forms/src/model.ts`!) could be exposed.

**Comment by akloeber** [comment]
In addition to that it would be possible to add a dedicated type for the status (see #28047):
```ts
import {VALID, INVALID, PENDING, DISABLED} from '@angular/forms';
export declare type FormControlStatus = VALID | INVALID | PENDING | DISABLED;
// or
export enum FormControlStatus {VALID = VALID, INVALID = INVALID, PENDING = PENDING, DISABLED = DISABLED}
```
That would make the available status self-documenting in TypeScript and also help with auto-completion in IDEs.

**Comment by coderbase-it** [comment]
Any update on it ? Exposing this kind of enum would be create for avoiding magic string like 
status === 'VALID'

**Comment by Benny739** [comment]
Any news on this?

**Comment by Wilt** [comment]
Sometimes you wonder whether angular is a TypeScript framework...
Version 9, but basic stuff like this is not typed.
I try always to be positive, but this is very disappointing...

**Comment by FDiskas** [comment]
And one year later... :D

**Comment by hohnzy** [comment]
And another six months later....

**Comment by angelaki** [comment]
Come  oooon ... ;)

**Comment by dylhunn** [comment]
This was fixed in Angular 13.0.
